redaxo/src/core/lib/sql/sql.php

```diff
@@ -829,6 +829,30 @@ public function insert()
         return $this;
     }
 
+    public function insertOrUpdate()
+    {
+        if ($this->records) {
+            return $this->setMultiRecordQuery('INSERT', true);
+        }
+
+        // hold a copies of the query fields for later debug out (the class property will be reverted in setQuery())
+        $tableName = $this->table;
+        $values = $this->values;
+
+        $onDuplicateKeyUpdate = $this->buildOnDuplicateKeyUpdate(array_keys(array_merge($this->values, $this->rawValues)));
+        $this->setQuery(
+            'INSERT INTO ' . $this->escapeIdentifier($this->table) . ' SET ' . $this->buildPreparedValues() . ' ' . $onDuplicateKeyUpdate,
+            $this->values
+        );
+
+        // provide debug infos, if insert is considered successfull, but no rows were inserted.
+        // this happens when you violate against a NOTNULL constraint
+        if ($this->getRows() == 0) {
+            throw new rex_sql_exception('Error while inserting into table "' . $tableName . '" with values ' . print_r($values, true) . '! Check your null/not-null constraints!', null, $this);
```

Comment:
@staabm Weißt du noch, wann das genau vorkommt? Ist hier nur eine Kopie von `insert()`, wo du das Verhalten eingefügt hattest:
https://github.com/redaxo/redaxo/commit/c6c3c1091b252199b68240feec3f554c4d4782c8

Bei mir, wenn ich in eine `NOT NULL`-Spalte versuche `NULL` einzufügen, dann bricht PDO bereits mit einer Exception ab.

Reply:
Ich denke dass passiert wenn man keinen mysql strict mode verwendet

Reply:
strict mode schalten wir eigentlich ab:
https://github.com/redaxo/redaxo/blob/master/redaxo/src/core/lib/sql/sql.php#L82

Oder macht `sql_mode=""` was anderes?

Aber an unterschiedlichen mysql-versionen könnte es natürlich theoretisch liegen.

Reply:
Strict mode abschalten haben wir aber erst später eingebaut, als dass diese fehlercondition vorhanden war

Reply:
D.h. Ggf. Ist das heute dead code

Reply:
Ich glaube ich lasse es dann aber trotzdem erst mal einfach drin, bzw. füge es hier in der neuen Methode mit ein.
Kann man dann ggf. separat sich mal genauer anschauen, ob das noch notwendig ist, oder nicht.